### PR TITLE
LMB-131: Cleanup form instance components

### PR DESCRIPTION
**Pull request by JonasVanHoof**
## Description

We have two form components to edit and create entities. The logic in these components are similar on a lot of places. We want to cleanup these components so they are still easy to understand but do not have so many duplicate code.

## How to test

1. Create a new entity with a form (/forms route)
2. Update an entity that is being updated with a form (e.g. correct mandataris)

### CODE REVIEW

File: app/components/form/new-instance.js [hunk 1]
```diff
@@ -38,21 +33,12 @@ export default class NewInstanceComponent extends Component {
...
-    return this.formInfo !== null;
-  }
-
-  get isSaving() {
-    return this.save.isRunning;
+    this.setupNewForm.perform();
   }
 
   save = task({ keepLatest: true }, async () => {
     // TODO validation needs to be checked first before the form is actually saved
```
Comment (JonasVanHoof) [inline app/components/form/new-instance.js:40]: Is this comment still accurate with a validation running on L44?

File: app/services/form-repository.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        errorMessage:
+          'Het formulier werd niet correct opgeslagen. Probeer het later opnieuw.',
+      };
+    }
+
+    return {
+      id: id,
+      errorMessage: null,
+    };
+  }
```
Comment (JonasVanHoof) [inline app/services/form-repository.js:77]: Not happy with these objects. Make it that you only return at the end of the method

File: app/utils/is-valid-form.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { validateForm } from '@lblod/ember-submission-form-fields';
+
+export function isValidFormFromFormInfo(formInfo) {
```
Comment (ruben-bottu) [inline app/utils/is-valid-form.js:3]: This function name is a bit confusing. Maybe something like `isValidFormInfo` ?

File: app/services/form-repository.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        }),
+      }
+    );
+
+    await timeout(RESOURCE_CACHE_TIMEOUT);
+
+    return await this._handleUpdateFetchhResult(result);
+  }
+
+  async _handleCreateFetchhResult(result) {
```
Comment (ruben-bottu) [inline app/services/form-repository.js:54]: Typos! `_handleCreateFetchhResult` and `_handleUpdateFetchhResult` have too many h'es
besides that, I would rename it to `handleCreateResponse` and `handleUpdateResponse`

**Review by ruben-bottu — APPROVED** [review]
Left a few comments, but overall looks good

File: app/utils/is-valid-form.js (hunk 3 above)
Comment (JonasVanHoof) [reply app/utils/is-valid-form.js]: Renamed to `isValidForm`

File: app/services/form-repository.js (hunk 4 above)
Comment (JonasVanHoof) [reply app/services/form-repository.js:54]: Is it validating something? Ill change it to `_handleCreateResult` and `_handleUpdateResult`

**Pull request merged**